include/swift/AST/Decl.h

```diff
@@ -3401,6 +3401,15 @@ class EnumDecl final : public NominalTypeDecl {
 /// to get the declared type ("Complex" in the above example).
 class StructDecl final : public NominalTypeDecl {
   SourceLoc StructLoc;
+  // We import C++ class template instantiations as non-generic structs
+  // with a name prefixed with `__CxxTemplateInst`. However for proper
+  // serialization we need to have an access to the bound generic type
+  // that would have produced this instantiation. This field contains
+  // that type.
```

Comment:
"For proper serialization" does not explain the difficultly, only asserts that this is the right solution. Could you add a bit more of an explanation why we need to serialize the bound generic?

Specifically, you should say that when we deserialize a reference to a class template instantiation imported in Swift, we need to have both the Swift struct in the Swift AST and the Clang class template specialization in the Clang AST. Deserialization process will create the Swift struct decl, so we are covered there. However, the corresponding Clang class template specialization is not guaranteed to already exist in any of the Clang modules that we imported so far. Therefore, we need to either instantiate the template during deserialization (the currently implemented solution), or serialize the class template specializations that Swift typechecker asks Clang to create (the ideal long-term solution that's not implemented now).

Reply:
Done, ptal :)